### ISSUE TITLE
(feat) Robot: Add setters to modify sleep duration.

### ISSUE THREAD

**Issue by hastebrot** [TestFX/TestFX#256]

**Comment by hastebrot** [comment]
So, for instance `WriteRobotImpl` requires to have a setter method for the configuration value (`writeIntervalInMillis`) in order to test the class. The `WriteRobot` interface should not contain this setter, because I think it is implementation specific.

I thought about adding a `ConfigRegistry` to `FxRobotContext` and allow something like `configRegistry.set(WriteRobotImpl.WRITE_INTERVAL, 25)` or `configRegistry.set(KeyEvent.KEY_TYPED, 25)`. But it introduces too many object dependencies.

**Comment by argonnet** [comment]
I don't understand this : 
>  The WriteRobot interface should not contain this setter, because I think it is implementation specific.

I think that in the fact of writing text there is two things : 
- The text you need to write
- The speed at which you want to write it

It seems right to specify it in the interface. Regardless the implementation, it seems correct (for me) to specify the speed you do it.  


As the writing speed is impacting only the writing of a text (and not a char), could we add a overload of the write method with the speed writing : 

`public void write(String text, long writeIntervalInMillis); `

It's a bit heavy for the final user but by this way no need to add setter in the interface. And the final user will be able to decide which part of code he will set the writing speed.

**Comment by hastebrot** [comment]
Thinking this one step further, this would mean, we also need to add `mouseMoveDuration` and `waitAfterDoubleClick` as parameters to `FxRobot::doubleClickOn()`. And there could be more configuration parameters, that we want to access, e.g. `boolean onlyVisibleNodes` as additional parameter.

**Comment by argonnet** [comment]
I haven't thought to this.

What do you think of creating a public Singleton class that contain all needed properties of this type with setter and getter. It could be a sort of TestFX configuration class ?

**Comment by hastebrot** [comment]
>What do you think of creating a public Singleton class that contain all needed properties

Yeah, I already thought about this [1] (`ConfigRegistry` in a previous comment). But I'm unsure.

https://github.com/TestFX/TestFX/issues/256#issuecomment-205292549

**Comment by argonnet** [comment]
Ok. 

When you speak of `ConfigRegistry`, you want to use ane existing librairy to do it ? 
And this is why you say that it would add to much dependencies ? Or you speak of the dependencies for classes of the project to the Singleton ?

Because, I think of a Java class like that : 

```
public class TestFXConfig{
	
	private long sleepAfterCharacterInMillis;
	
	public TestFXConfig(){
		//Default Value
		sleepAfterCharacterInMillis = 25
	}
	
	public long getSleepAfterCharacterInMillis(){return sleepAfterCharacterInMillis;}
	public void setSleepAfterCharacterInMillis(long value){this.sleepAfterCharacterInMillis = value;}
}
```

No need of external librairy. I know that many classes will be pointing on it but I don't think this is bad.

**Comment by hastebrot** [comment]
>Or you speak of the dependencies for classes of the project to the Singleton ?

This.

So far I thing there should be (at least) two ways to define the flags (registry settings):

- via java system properties for the whole test run.
- via field in `ConfigRegistry` for a single test class.

The reason, why I want to define the fields only for a single test class (second case) is, because I have to set the field somewhere in a specific class.

**Comment by hastebrot** [comment]
FEST-Swing has this:

~~~groovy
def fixture = new FrameFixture(robot, frame)
fixture.robot.settings().delayBetweenEvents(delayMilliseconds)
fixture.robot.settings().componentLookupScope(ComponentLookupScope.ALL)
~~~

**Comment by argonnet** [comment]
It seems easy to use like that :-)

You want to be able to configure this options for each class you want to test ?

```
ClassA -> Speed set to 10 ms
ClassB -> Speed set to 15 ms 
```

This is why you want to use the `ConfigRegistry `?
